runtime/queries/rust/indents.toml

```diff
@@ -1,4 +1,9 @@
-indent = [
+[indent]
+all = [
+ { parent_kind_in = ["assignment_expression", "compound_assignment_expr"], field_name_in = ["right"]},
+ { parent_kind_in = ["let_declaration"], field_name_in = ["value"] }
+]
+tail = [
```

Comment:
I've read through both the examples the docs and the original issue, but I don't understand the need for `all` vs `tail`.

The point is to make:

```rust
let (
    a_long_variable_name_in_this_tuple,
    b_long_variable_name_in_this_tuple,
    c_long_variable_name_in_this_tuple,
    d_long_variable_name_in_this_tuple,
    e_long_variable_name_in_this_tuple,
): (usize, usize, usize, usize, usize) = //<- cursor here
```

`<Enter>` produce indentation level == 1 here?

Reply:
Yes, basically. `indent.tail` produces the behavior of the old `indent` option (indenting everything except for the first line of a node). `indent.all` indents the whole node including the first line, similar to how the old `outdent` behaves. Here we need it because the whole RHS of a let declaration should be indented, including the first line. Additionally, `indent` and `outdent` are symmetric in the new version, so we don't need any special cases for them and for everything else that the indent code might handle in the future (comments, alignment, etc.).

Your example doesn't work because of a limitation of tree-sitter: In that situation there isn't a value for the `let_declaration` yet, so we don't know that we should indent it. The best solution here is probably an optional auto-indent feature that updates the indent as soon as tree-sitter parses the code correctly.
Pressing `<Enter>` in this situation does produce the correct indent (1):
```rust
let (
    a_long_variable_name_in_this_tuple,
    b_long_variable_name_in_this_tuple,
    c_long_variable_name_in_this_tuple,
    d_long_variable_name_in_this_tuple,
    e_long_variable_name_in_this_tuple,
): (usize, usize, usize, usize, usize) = |(0,0,0,0,0);
// Cursor at | in the line above
```